### PR TITLE
Remove sentiment extension

### PR DESCRIPTION
**Pull request by thomashacker**
## Description

This `PR` removes the `sentiment` extension attribute from the `Token`, `Span`, `Doc` and `Lexeme` class.

The `sentiment` extension is tied to one particular model of sentiment analysis that is kind of old and limited in applications. In general it makes sense to make custom field support better where there's not a performance impact rather than having more special-case fields built in.

### Types of change

Remove feature and change to documentation

## Checklist

- [X] I confirm that I have the right to submit this contribution under the project's MIT license.
- [X] I ran the tests, and all new and existing tests passed.
- [X] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by polm — APPROVED** [review]
Looks good to me! I'm not sure if we should get other people to weigh in on this or not?

**Comment by adrianeboyd** [comment]
Additional files aren't included in the package by default, so you can copy the bytestring directly into the test file instead. Try to make the example as short as you can, maybe just one token + `.sentiment` and no additional annotation. Alternatively, you could adjust the settings in `MANIFEST.in` to include this file, but I think this can be made short enough to include directly.

Separate from this PR, we probably need to think about how to extend the test suite with more examples like this without bloating the final `spacy` package too much. We were thinking about having two test suites, one that's packaged with `spacy.tests` and a separate one with larger/slower tests that run in the CI but not with `pytest --pyargs spacy`.

File: website/docs/usage/rule-based-matching.md [hunk 1]
```diff
@@ -777,6 +777,8 @@ whitespace, making them easy to match as well.
 from spacy.lang.en import English
 from spacy.matcher import Matcher
 
+Doc.set_extension("sentiment", default=0.0)
```
Comment (adrianeboyd) [inline website/docs/usage/rule-based-matching.md:780]: Both the examples here are missing the `Doc` import.

File: spacy/tests/doc/test_doc_api.py [hunk 2]
```diff
@@ -990,3 +990,11 @@ def test_doc_spans_setdefault(en_tokenizer):
     assert len(doc.spans["key2"]) == 1
     doc.spans.setdefault("key3", default=SpanGroup(doc, spans=[doc[0:1], doc[1:2]]))
     assert len(doc.spans["key3"]) == 2
+
+def test_doc_sentiment_from_bytes_v3_to_v4():
```
Comment (adrianeboyd) [inline spacy/tests/doc/test_doc_api.py:994]: Can you reformat this file?

File: spacy/tests/README.md [hunk 3]
```diff
@@ -10,12 +10,16 @@ Tests for spaCy modules and classes live in their own directories of the same na
 
 ## Table of contents
 
-1. [Running the tests](#running-the-tests)
-2. [Dos and don'ts](#dos-and-donts)
-3. [Parameters](#parameters)
-4. [Fixtures](#fixtures)
-5. [Helpers and utilities](#helpers-and-utilities)
-6. [Contributing to the tests](#contributing-to-the-tests)
+- [spaCy tests](#spacy-tests)
```
Comment (adrianeboyd) [inline spacy/tests/README.md:13]: What happened to the formatting here? Was this intentional?

File: spacy/tests/doc/test_span.py [hunk 4]
```diff
@@ -304,32 +304,6 @@ def test_span_similarity_match():
         assert span1.similarity(doc) == 0.0
         assert span1[:1].similarity(doc.vocab["a"]) == 1.0
 
-
```
Comment (adrianeboyd) [inline spacy/tests/doc/test_span.py:307]: Can you reformat this file, I think there should be one more blank line.

File: spacy/tests/README.md (hunk 3 above)
Comment (thomashacker) [reply spacy/tests/README.md]: Oh, I didn't realize, `prettier` formatted the table of contents automatically to this. I've replaced the old table and saved without formatting.

**Review by svlandeg — APPROVED** [review]
Looks good to me! I'll leave the final decision to merge or not with Adriane so that she can check that all her reviewer comments have been addressed.

**Comment by adrianeboyd** [comment]
Looks good!

**Pull request merged**